compiler/resolution/callDestructors.cpp

```diff
@@ -438,8 +542,44 @@ void ReturnByRef::transformMove(CallExpr* moveExpr)
 
   Expr*     lhs      = moveExpr->get(1);
   CallExpr* callExpr = toCallExpr(moveExpr->get(2));
+  CallExpr* copyExpr = NULL;
   Symbol*   useLhs   = toSymExpr(lhs)->var;
   Symbol*   refVar   = newTemp("ret_to_arg_ref_tmp_", useLhs->type->refType);
+  FnSymbol* fn       = callExpr->isResolved();
+
+  if (fn == NULL)
```

Comment:
I like the second form you have there, its clearer to me what is going on at least.